### PR TITLE
Add SD3

### PR DESCRIPTION
**Pull request by lockwo**
This is a first draft of the SD3 algo (https://arxiv.org/pdf/2010.09177.pdf, NeurIPS'20). It is very similar to TD3, but adjusts the Q function with a softmax operation (and has some minor implementation changes). Here are some example runs: 
image
image
Not a lot of runs, but I'm looking to get a few more. Haven't done any sweeps yet.

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Hey @lockwo this looks pretty cool. Maybe it's also worth trying out a style similar to https://github.com/vwxyzjn/vectorized-value-methods/blob/direct-returns/vrtd3_continuous_action.py?

**Comment by lockwo** [comment]
I started drafting a similar style implementation (https://github.com/lockwo/vectorized-value-methods/blob/returns/vrsd3_continuous_action.py), it currently doesn't learn very well. Beyond the basic lack of hyperparameter adjustment, there is the difficulty of properly adjusting the softmax operation on the returns (just like the challenge of adding the entropy to the SAC from returns).
